include/mega/logging.h

```diff
@@ -109,41 +119,223 @@ enum LogLevel {
 };
 
 // Output Log Interface
-class Logger {
+class Logger
+{
 public:
+    virtual ~Logger() = default;
     virtual void log(const char *time, int loglevel, const char *source, const char *message) = 0;
 };
 
 typedef std::vector<std::ostream *> OutputStreams;
 
 class OutputMap : public std::array<OutputStreams, unsigned(logMax)+1> {};
 
-class SimpleLogger {
+class SimpleLogger
+{
     enum LogLevel level;
     std::ostringstream ostr;
     std::string t;
     std::string fname;
 
     std::string getTime();
 
+    std::array<char, 256> mBuffer; // used in performance mode (stack-allocated since SimpleLogger is normally stack-allocated)
+    std::array<char, 256>::iterator mBufferIt; // used in performance mode
+
     // logging can occur from multiple threads, so we need to protect the lists of loggers to send to
     // though the loggers themselves are presumed to be owned elsewhere, and the pointers must remain valid
     // actual output to the loggers is not synchronised (at least, not by this class)
     static std::mutex outputs_mutex;
     static OutputMap outputs;
     static OutputStreams getOutput(enum LogLevel ll);
 
+    static bool performanceMode;
```

Comment:
= false;
missing?

Reply:
It's initialized in the cpp. But I agree, we should make this a compile time decision so we avoid the runtime checks (Matt also agrees). How about we call this: `ENABLE_LOG_PERFORMANCE_MODE` to indicate it's related to logging?